### ISSUE TITLE
bootstrap 4 support

### ISSUE THREAD

**Issue by acrobat** [eternicode/bootstrap-datepicker#1670]
Bs4 is still in early alpha but when it reaches beta (no api changes after beta) we should take a look into adding a bs4 stylesheet and possibly deprecating bs2 support

**Comment by hebbet** [comment]
> possibly deprecating bs2 support

would drop support for bs2 with the first version supporting bs4

**Comment by acrobat** [comment]
Yes you are right @hebbet, bs2 is not supported by the twbs team so we should drop it too when moving to bs4

**Comment by EugenMayer** [comment]
Is there any progress on this side at the moment? We are using bootstrap4-alpha3 pretty heavily already and i would suggest it could make sense migrating this also? We have been using bootstrap-datepicker with bootstrap2 and would love continuing to do so.

What are the expected working-areas to adjust the datebicker? are those mostly theme / css related things?

**Comment by acrobat** [comment]
I haven't looked into detail on what should/must change, but my first idea is that we would need to switch some classes and provide a bs4 stylesheet

**Comment by EugenMayer** [comment]
Well if you guys could guide me a little bit, we could try to do some first attempts here to see, how "low hanging fruits" this topic might be and what concrete issues would arise. What do you think?

**Comment by acrobat** [comment]
@EugenMayer all help is welcome! Maybe you could just load up a bs3 datepicker in the new bs4 style and see what styles are wrong and start making a list of classes/styles (in a separate issue) we should change. That way we have already an idea on how much work this will be

After that we can start on a plan what steps we should take (the introduction of themes is one of those)

**Comment by EugenMayer** [comment]
Perfect, will do that and report back. We will try to see what we can fix and show off those things using pull requests to see, how things work out

**Comment by acrobat** [comment]
That's great @EugenMayer, thanks!

**Comment by Azaret** [comment]
This will be done when the backends stuff is done.
Not that in my hand I won't create a backend that support BS4, tho I'll create on for BS2, mainly because I do still use BS2 on some projects.

**Comment by johan-bjareholt** [comment]
Since others are referencing this issue mentioning that it works but isn't officially supported and bootstrap4 has been stable for quite some time, when do you expect to make it officially support bootstrap4?

**Comment by EugenMayer** [comment]
bootstrap is no close to stable - still :)

**Comment by alvipeo** [comment]
I gave up on bootstrap completely about 6 months ago and happy! Use semantic-ui or material.

**Comment by johan-bjareholt** [comment]
Wow, I didn't know it was this bad. I was using bootstrap-vue which is the only vue2 plugin which uses bootstrap and it used bootstrap4 so i assumed it was stable.

**Comment by ndabAP** [comment]
@johan-bjareholt, I'm in the exact same situation. Using bootstrap-vue and looking for a Bootstrap 4 datetimepicker.

**Comment by ieugen** [comment]
Bootstrap 4 has just gone beta today. Happy coding!

**Comment by tavuntu** [comment]
Hey guys, how stable is BS4 support right now? (for the beta version)

**Comment by acrobat** [comment]
At this point we should be able to start work on bs4 support ideally with some sort of theme support so we can easily support bs3 and bs4 (and even other themes if necessary).

I don't have the time (and js knowledge) to start work on this feature. Therefor I've opened #2252 to request some extra helper from active JS developers. This way we can, hopefully, get more feature implemented and bugs fixed! So any idea's and development help are welcome!

**Comment by ttskch** [comment]
maybe same as #1574

**Comment by aasutossh** [comment]
Let's specify on the readme that BS4 isn't supported so that people like me don't have to beat their head. It says it needs BS2.0.4+ on the RTD, I assumed that it would support BS4.
